ads/google/a4a/performance.js

```diff
@@ -194,36 +198,65 @@ export class GoogleAdLifecycleReporter extends BaseLifecycleReporter {
    * @override
    */
   sendPing(name) {
-    this.emitPing_(this.buildPingAddress_(name));
+    this.emitPing_(this.buildPingAddress_(name, this.extraVariables_));
   }
 
   /**
+   * @param {string} variable
+   * @param {string|number} value
+   * @override
+   */
+  setPingVariable(variable, value) {
```

Comment:
Seems you could make this concrete within the base class assuming all sub-classes would do the same thing (write to some object)